src/components/radio/radio.spec.ts

```diff
@@ -1,370 +1,398 @@
+      let changeCount = 0;
+      radioInstances[0].change.subscribe(() => {
+        changeCount++;
+
+        // One event for checked, one for unchecked.
+        if (changeCount == 2) {
+          promiseCompleter.resolve();
+        }
+      });
+
+      radioInstances[0].checked = true;
+      fixture.detectChanges();
+
+      radioInstances[0].checked = false;
+      fixture.detectChanges();
+
+      return promiseCompleter.promise;
     });
 
-    it('should be focusable', (done: () => void) => {
-      builder
-        .overrideTemplate(TestApp, '<md-radio-button></md-radio-button>')
-        .createAsync(TestApp)
-        .then(fixture => {
-          let button = fixture.debugElement.query(By.css('md-radio-button'));
-          let input = button.query(By.css('input'));
-
-          fixture.detectChanges();
-          expect(button.nativeElement.classList.contains('md-radio-focused')).toBe(false);
-
-          let event = createEvent('focus');
-          input.nativeElement.dispatchEvent(event);
-          fixture.detectChanges();
-          expect(button.nativeElement.classList.contains('md-radio-focused')).toBe(true);
-
-          event = createEvent('blur');
-          input.nativeElement.dispatchEvent(event);
-          fixture.detectChanges();
-          expect(button.nativeElement.classList.contains('md-radio-focused')).toBe(false);
-        }).then(done);
+    it('should emit a change event from the radio group', () => {
+      expect(groupInstance.value).toBeFalsy();
+
+      // TODO(jelbourn): this *should* work with async(), but fixture.whenStable currently doesn't
+      // know to look at pending macro tasks.
+      // See https://github.com/angular/angular/issues/8389
+      // As a short-term solution, use a promise (which jasmine knows how to understand).
+      let promiseCompleter = new PromiseCompleter();
+      let changeCount = 0;
+      groupInstance.change.subscribe(() => {
+        changeCount++;
+
+        // One event for checked, one for unchecked.
+        if (changeCount == 2) {
+          promiseCompleter.resolve();
+        }
+      });
+
+      groupInstance.value = 'fire';
+      fixture.detectChanges();
+
+      groupInstance.value = 'water';
+      fixture.detectChanges();
+
+      return promiseCompleter.promise;
     });
-  });
 
-  describe('MdRadioDispatcher', () => {
-    let builder: TestComponentBuilder;
-    let dispatcher: MdRadioDispatcher;
+    it('should focus individual radio buttons', () => {
+      let nativeRadioInput = <HTMLElement> radioNativeElements[0].querySelector('input');
 
-    beforeEach(inject([TestComponentBuilder], (tcb: TestComponentBuilder) => {
-      builder = tcb;
-      dispatcher = new MdRadioDispatcher();
-    }));
+      expect(nativeRadioInput.classList).not.toContain('md-radio-focused');
 
-    it('notifies listeners', () => {
-      let notificationCount = 0;
-      const numListeners = 2;
+      dispatchFocusChangeEvent('focus', nativeRadioInput);
```

Comment:
I tried it and it does not seem to work. Added a TODO to write an e2e test that uses real focus.